public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -208,11 +208,39 @@ export default class Datasource extends DataSourceWithBackend<AzureMonitorQuery,
     return this.templateSrv.getVariables().map((v) => `$${v.name}`);
   }
 
+  trackQueries(
+    queries: AzureMonitorQuery[],
+    dashboardId?: string,
+    orgId?: number,
+    userId?: number,
+    grafanaVersion?: string
+  ) {
+    reportInteraction('grafana_azuremonitor_dashboard_loaded', {
```

Comment:
**Nit:** maybe I would add `ds_` in the name, so it's easier to group for events coming from datasources later?

E.g.: `grafana_ds_azuremonitor_dashboard_loaded`

Maybe I am wrong though as I don't know too much about event filtering in Rudderstack :)